scheduler/src/cook/monitor.clj

```diff
@@ -145,18 +146,30 @@
   (run!
     (fn [[user stats]]
       (run! (fn [[type amount]]
-              (-> [state user (name type) (str "pool-" pool-name)]
+              (do
+                (-> [state user (name type) (str "pool-" pool-name)]
                   counters/counter
-                  (set-counter! amount)))
+                  (set-counter! amount))
+                ; Metrics need to be pre-registered in prometheus, so only record them if the metric exists
+                ; Log a warning otherwise so that we know to add a metric if we add a new resource type.
+                (if (and (contains? prometheus/resource-metric-map state) (contains? (prometheus/resource-metric-map state) type))
```

Comment:
Am I missing something? It looks like `(contains? prometheus/resource-metric-map state)` is duped, but gh's syntax highlighting is gaslighting me...